test/helpers.lua

```diff
@@ -269,43 +269,39 @@ function module.check_logs()
     table.concat(runtime_errors, ', ')))
 end
 
-function module.iswin()
-  return package.config:sub(1,1) == '\\'
-end
-
--- Gets (lowercase) OS name from CMake, uname, or "win" if iswin().
-module.uname = (function()
-  local platform = nil
-  return (function()
```

Comment:
This closed over the `platform` local so that we don't keep shelling out for every call. Not a big deal but why reverse that?

Reply:
I thought it looked like a mistake, I didn't even know you could do that in lua.

Reply:
We're using a function as a glorified constant, it's obnoxiously clever in the worst way.

Reply:
If we want to avoid the ` = (function() ... end)()` pattern (which I agree is a little _too_ clever), just use a `do ... end` block:

```lua
do
  local platform = nil
  function module.uname()
    ...
  end
end
```